source/extensions/access_loggers/filters/cel/config.cc

```diff
@@ -0,0 +1,65 @@
+#include "source/extensions/access_loggers/filters/cel/config.h"
+
+#include "envoy/extensions/access_loggers/filters/cel/v3/cel.pb.h"
+
+#include "source/extensions/access_loggers/filters/cel/cel.h"
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
+
+namespace Envoy {
+namespace Extensions {
+namespace AccessLoggers {
+namespace Filters {
+namespace CEL {
+
+Envoy::AccessLog::FilterPtr CELAccessLogExtensionFilterFactory::createFilter(
+    const envoy::config::accesslog::v3::ExtensionFilter& config, Runtime::Loader&,
+    Random::RandomGenerator&) {
+
+  // TODO: use factory_context validation. likely needs update to createFilter
+  // signature to pass in validation visitor.
+  auto factory_config = Config::Utility::translateToFactoryConfig(
+      config, Envoy::ProtobufMessage::getNullValidationVisitor(), *this);
+
+#if defined(USE_CEL_PARSER)
+  envoy::extensions::access_loggers::filters::cel::v3::ExpressionFilter cel_config =
+      *dynamic_cast<const envoy::extensions::access_loggers::filters::cel::v3::ExpressionFilter*>(
+          factory_config.get());
+
+  auto parse_status = google::api::expr::parser::Parse(cel_config.expression());
+  if (!parse_status.ok()) {
+    throw EnvoyException("Not able to parse filter expression: " +
+                         parse_status.status().ToString());
+  }
+
+  return std::make_unique<CELAccessLogExtensionFilter>(getOrCreateBuilder(),
+                                                       parse_status.value().expr());
+#else
+  return nullptr;
```

Comment:
I think we should throw an exception here. If I understand correctly, this is a case of someone configuring an extension that requires CEL on a build that doesn't support CEL and it seems like that should be a config-time error.

Further, looking at the code that calls this, there are definitely cases where the result of this call is blindly added to a list of filters (e.g. in the `OperatorFilter` constructor), which seems like it'll probably just cause a crash later (in say, `OrFilter::evaluate`).

Reply:
Updated to throw an exception.